test/parser/statement/Function.test.js

```diff
@@ -12,6 +12,33 @@ describe("parser", () => {
     });
 
     describe("function declarations", () => {
+        it('recovers when using `end sub` instead of `end function`', () => {
+            const { tokens } = brs.lexer.Lexer.scan(`
+                function Main()
+                    print "Hello world"
+                end sub
+                
+                sub DoSomething()
+                
+                end sub 
+            `);
+            let { statements, errors } = parser.parse(tokens);
+            expect(errors.length).toBe(1);
+            expect(errors[0].location).toMatchObject({
```

Comment:
Like in #203, consider using a second snapshot here!  The location seems slightly less important than the error's message, and we can check both at the same time with a second snapshot :camera_flash: